app/auth/views/proton.py

```diff
@@ -166,5 +167,4 @@ def check_status_code(response: requests.Response) -> requests.Response:
     if res.redirect:
         return after_login(res.user, res.redirect, login_from_proton=True)
 
-    next_url = session.get("oauth_next")
     return after_login(res.user, next_url, login_from_proton=True)
```

Comment:
To avoid having to pass the next_url to handle_link we could do:

```python
redirect_url = next_url or res.redirect
after_login(res.user, redirect_url, login_from_proton=True)
```